paymentsheet/src/main/java/com/stripe/android/paymentsheet/viewmodels/PrimaryButtonUiStateMapper.kt

```diff
@@ -31,31 +30,25 @@ internal class PrimaryButtonUiStateMapper(
     private val onClick: () -> Unit,
 ) {
 
-    private fun screenAndButtonFlow(): StateFlow<Pair<PaymentSheetScreen, BuyButtonState>> {
-        return currentScreenFlow.flatMapLatestAsStateFlow { screen ->
-            screen.buyButtonState.mapAsStateFlow { buyButtonState ->
-                screen to buyButtonState
-            }
-        }
-    }
-
     fun forCompleteFlow(): Flow<PrimaryButton.UIState?> {
         return combine(
-            screenAndButtonFlow(),
+            currentScreenFlow,
             buttonsEnabledFlow,
             amountFlow,
             selectionFlow,
             customPrimaryButtonUiStateFlow,
             cvcCompleteFlow
-        ) { (screen, buyButtonState), buttonsEnabled, amount, selection, customPrimaryButton, cvcComplete ->
-            customPrimaryButton ?: PrimaryButton.UIState(
-                label = buyButtonState.buyButtonOverride?.label ?: buyButtonLabel(amount),
-                onClick = onClick,
-                enabled = buttonsEnabled && selection != null &&
-                    cvcRecollectionCompleteOrNotRequired(screen, cvcComplete, selection),
-                lockVisible = buyButtonState.buyButtonOverride?.lockEnabled ?: true,
-            ).takeIf { buyButtonState.visible }
-        }
+        ) { screen, buttonsEnabled, amount, selection, customPrimaryButton, cvcComplete ->
+            screen.buyButtonState.map { buyButtonState ->
```

Comment:
I expected this to happen in the combine.

Ie `currentScreenFlow.map { ... },`

But not blocking.